be/Controller/RemoveGameCtrl.php

```diff
@@ -62,30 +61,29 @@ private function removeLastGame(
 
     private function removeLastGameIfPossibleInDb(string $code): array
     {
-        $user = $this->userRepository->requireUserByCode($code);
-
-        $userNid = $user->getUserNid();
-
         try {
-            $lastGame = $this->gameRepository->requireLastGame($userNid);
+            $user = $this->userRepository->requireUserByCode($code);
         } catch (NoResultException $e) {
             return [
                 'removed' => false,
-                'warning' => 'Player has no games',
+                'warning' => 'Player does not exist',
             ];
         }
 
-        $opponentUser = $this->getOpponentFromGame($lastGame, $userNid);
+        $userNid = $user->getUserNid();
 
         try {
-            $opponentLastGame = $this->gameRepository->requireLastGame($opponentUser->getUserNid());
+            $lastGame = $this->gameRepository->requireLastGame($userNid);
         } catch (NoResultException $e) {
```

Comment:
No i tak ogólnie się zastanawiam czy w kodzie dobre jest korzystanie z try-catch ?
Bo teoretycznie to że ktoś nie ma lastGame jest jak najbardzie legalne. Zastanawiam się czy to powinno być robione przez require... który robi throw. Czy te funkcjie require... nie powinno się tworzyć wtedy jak wiemy że coś napewno musi istnieć (i wtedy jak nie istnieje to jest prawdziwy błąd i wtedy nie trzeba otaczać tu try catchem bo właśnie ma się wywalić). To takie ogólne przemyślenia, nie wiem co jest dobrą praktyką

Reply:
Takie ogólne przemyślenia jak najbardziej cenię i dobrze jest rozpoczynać rozmowy na ich temat :).
W sumie, to możesz mieć rację. W tym akurat przypadku rzeczywiście można logiką sterować chociażby przy pomocy warunku typu `if ($lastGame === null) { }`. Powiem Ci, że często mam z tym problem, i nigdy nie znalazłem jakiegoś artykułu, który by tłumaczył tego typu sytuacje.